core/src/scene/sceneLoader.h

```diff
@@ -37,59 +31,87 @@ struct StyleUniform {
-                             const std::vector<SceneUpdate>& updates);
-    static void applyGlobals(Node root, Scene& scene);
-
-    /*** all public for testing ***/
-
-    static void loadBackground(Node background, const std::shared_ptr<Scene>& scene);
-    static void loadSource(Platform& platform, const std::string& name,
-                           const Node& source, const Node& sources, const std::shared_ptr<Scene>& scene);
-    static void loadSourceRasters(Platform& platform, std::shared_ptr<TileSource>& source, Node rasterNode,
-                                  const Node& sources, const std::shared_ptr<Scene>& scene);
-    static void loadTexture(Platform& platform, const std::pair<Node, Node>& texture, const std::shared_ptr<Scene>& scene);
-    static void loadLayer(const std::pair<Node, Node>& layer, const std::shared_ptr<Scene>& scene);
-    static void loadLight(const std::pair<Node, Node>& light, const std::shared_ptr<Scene>& scene);
-    static void loadCameras(Node cameras, const std::shared_ptr<Scene>& scene);
-    static void loadCamera(const Node& camera, const std::shared_ptr<Scene>& scene);
-    static void loadStyleProps(Platform& platform, Style& style, Node styleNode, const std::shared_ptr<Scene>& scene);
-    static void loadMaterial(Platform& platform, Node matNode, Material& material, const std::shared_ptr<Scene>& scene, Style& style);
-    static void loadShaderConfig(Platform& platform, Node shaders, Style& style, const std::shared_ptr<Scene>& scene);
-    static void loadFont(Platform& platform, const std::pair<Node, Node>& font, const std::shared_ptr<Scene>& scene);
-    static SceneLayer loadSublayer(const Node& layer, const std::string& name, const std::shared_ptr<Scene>& scene);
-    static Filter generateFilter(Node filter, Scene& scene);
-    static Filter generateAnyFilter(Node filter, Scene& scene);
-    static Filter generateAllFilter(Node filter, Scene& scene);
-    static Filter generateNoneFilter(Node filter, Scene& scene);
-    static Filter generatePredicate(Node filter, std::string _key);
-    static bool getFilterRangeValue(const Node& node, double& val, bool& hasPixelArea);
-    /* loads a texture with default texture properties */
-    static std::shared_ptr<Texture> getOrLoadTexture(Platform& platform, const std::string& url, const std::shared_ptr<Scene>& scene);
-    static std::shared_ptr<Texture> fetchTexture(Platform& platform, const std::string& name, const std::string& url,
-                                                 const TextureOptions& options, const std::shared_ptr<Scene>& scene,
-                                                 std::unique_ptr<SpriteAtlas> _atlas = nullptr);
+    static SceneError applyUpdates(Node& config, const std::vector<SceneUpdate>& updates);
+
+
+    /// Global
+    static void applyGlobals(Node& config);
+
+
+    /// Scene
+    static void applyScene(const Node& sceneNode, Color& background, Stops& backgroundStops,
+
```

Comment:
Spa

cing

Reply:
It's intended for visual grouping - These sections could be moved to separate files but I didn't want to do this in this PR.

Reply:
Done